fiftyone/core/aggregations.py

```diff
@@ -1890,6 +1896,16 @@ def parse_result(self, d):
 
             if name in doc_fields:
                 field = doc_fields[name]
+            elif "." in _type:
+                _type, _cls = _type.split(".", 1)
```

Comment:
do we do any validation on field names? What if a user tries to add a dynamic field with a `.` in the name?

Reply:
If a user tried to set a field like `foo.bar` without first declaring `foo`, they'd see an error like this:

```
ValueError: Cannot infer an appropriate type for non-existent sample field 'foo' while defining embedded field 'foo.bar'
```